openraft/src/core/mod.rs

```diff
@@ -739,10 +738,9 @@ impl<'a, D: AppData, R: AppDataResponse, N: RaftNetwork<D>, S: RaftStorage<D, R>
         loop {
             if !self.core.target_state.is_leader() {
                 tracing::info!("id={} state becomes: {:?}", self.core.id, self.core.target_state);
+                // notify to all nodes DO NOT send replication event any more.
+                self.replication_rx.close();
```

Comment:
It seems like a single `return` would be enough: channel will be closed when dropped.

Although I am not sure if it's a good idea to do that 🤔 
What say you man?

Reply:
The difference between `drop` and `close` is that a closed receiver can still drain messages that the sender already ended to the channel buffer. In this case, `drop` is enough.
So you are totally right as always.